src/Sylius/Component/Core/Translation/TranslatableEntityLocaleAssigner.php

```diff
@@ -21,14 +22,27 @@
 
 final class TranslatableEntityLocaleAssigner implements TranslatableEntityLocaleAssignerInterface
 {
-    public function __construct(private LocaleContextInterface $localeContext, private TranslationLocaleProviderInterface $translationLocaleProvider)
-    {
+    public function __construct(
+        private LocaleContextInterface $localeContext,
+        private TranslationLocaleProviderInterface $translationLocaleProvider,
+        private ?CommandBasedContextCheckerInterface $commandBasedChecker = null
+    ) {
+        if ($this->commandBasedChecker === null) {
+            @trigger_error('Not passing 3rd argument is deprecated. Pass CommandBasedContextCheckedInterface', E_USER_DEPRECATED);
```

Comment:
```suggestion
            @trigger_error('Not passing CommandBasedContextCheckedInterface explicitly as the third argument is deprecated since 1.11 and will be prohibited in 2.0.', E_USER_DEPRECATED);
```